dGame/dComponents/SkillComponent.cpp

```diff
@@ -210,6 +213,30 @@ void SkillComponent::RegisterCalculatedProjectile(const LWOOBJID projectileId, B
 	this->m_managedProjectiles.push_back(entry);
 }
 
+bool SkillComponent::CastSkill(const uint32_t skillId, LWOOBJID target, const LWOOBJID optionalOriginatorID){
+	uint32_t behaviorId = -1;
+	// try to find it via the cache
+	const auto pair = SkillBehaviorCache.find(skillId);
+
+	// if it's not in the cache look it up and cache it
+	if (pair == SkillBehaviorCache.end()) {
+		auto skillTable = CDClientManager::Instance()->GetTable<CDSkillBehaviorTable>("SkillBehavior");
```

Comment:
I meant caching the table pointer itself :P
This caching also looks like a slight optimization.